include/swift/Remote/TypeInfoProvider.h

```diff
@@ -25,13 +26,22 @@ namespace remote {
 
 /// An abstract interface for providing external type layout information.
 struct TypeInfoProvider {
+  using IdType = uint64_t;
```

Comment:
Minor improvement: If the intention is to use an address in all implementations, we could use `void *` as the type, then it would take up less space on a 32-bit system.

Reply:
Otherwise this LGTM.